packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -376,5 +427,55 @@ export default createRule<Options, MessageId>({
       const type = getConstrainedTypeAtLocation(services, targetNode);
       return tsutils.isTypeFlagSet(type, ts.TypeFlags.VoidLike);
     }
+
+    function hasValidReturnType(
+      services: ParserServicesWithTypeInformation,
+      node:
+        | TSESTree.FunctionExpression
+        | TSESTree.ArrowFunctionExpression
+        | TSESTree.FunctionDeclaration
+        | null,
+    ): boolean {
+      if (node != null) {
+        const checker = services.program.getTypeChecker();
+        const functionTSNode = services.esTreeNodeToTSNodeMap.get(node);
+        const functionType =
+          (ts.isFunctionExpression(functionTSNode) ||
+          ts.isArrowFunction(functionTSNode)
+            ? getContextualType(checker, functionTSNode)
+            : services.getTypeAtLocation(node)) ??
+          services.getTypeAtLocation(node);
+
+        if (functionTSNode.type) {
+          for (const signature of tsutils.getCallSignaturesOfType(
+            functionType,
+          )) {
+            return !(
+              tsutils.isTypeFlagSet(
+                signature.getReturnType(),
+                ts.TypeFlags.Any,
+              ) ||
+              tsutils.isTypeFlagSet(
+                signature.getReturnType(),
+                ts.TypeFlags.Unknown,
+              )
```

Comment:
```suggestion
              tsutils.isTypeFlagSet(
                signature.getReturnType(),
                ts.TypeFlags.Any | ts.TypeFlags.Unknown,
              )
```

[Refactor] Let's do a bitwise OR of these flags instead of calling `isTypeFlagSet` twice?

Reply:
I did not understand that the isTypeFlagSet function utilizes bitwise operators, so I called it twice. thank you for telling me!